### ISSUE TITLE
[Clang] Add new warning for a == 10 ? false : true called -Wredundant-ternary.

### ISSUE THREAD

**Issue by BreadTom** [llvm/llvm-project#214715]
[GCC bug 126715](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=126715)
[Clang trunk godbolt](https://godbolt.org/z/hPov6qTGs) with -O3 -Wall -Wextra -Weverything
does not warn that the C/C++ ternary is redundant:

```c++
extern bool src (int a);

bool
src (int a)
{
  return a == 10 ? false : true;
}

```

This shows up in GCC pre [r17-2872 / 2f5ed170082f9f2401a0c1c4a18e538c0cf75013](https://gcc.gnu.org/git/?p=gcc.git;a=commit;h=2f5ed170082f9f2401a0c1c4a18e538c0cf75013).

Maybe call this warning -Wredundant-ternary and make it default in -Wall?

**Comment by EugeneZelenko** [comment]
See Clang-Tidy [simplify-boolean-expr](https://clang.llvm.org/extra/clang-tidy/checks/readability/simplify-boolean-expr.html) check.
